products.go

```diff
@@ -0,0 +1,129 @@
+package paypal
+
+import (
+	"fmt"
+	"net/http"
+)
+
+type (
+	ProductType     string
+	ProductCategory string
+
+	// Product struct
+	Product struct {
+		ID          string          `json:"id,omitempty"`
+		Name        string          `json:"name,omitempty"`
+		Description string          `json:"description"`
+		Category    ProductCategory `json:"category"`
+		Type        ProductType     `json:"type"`
```

Comment:
lets make these pointers and omit empty?

Reply:
`type` is a required parameter
Category is a string, not a good idea to use pointers with strings

Reply:
got it didnt realize type was required.